lib/vector-buffers/src/internal_events.rs

```diff
@@ -57,43 +66,60 @@ impl InternalEvent for BufferCreated {
 }
 
 pub struct BufferEventsReceived {
+    pub buffer_id: String,
     pub idx: usize,
     pub count: u64,
     pub byte_size: u64,
 }
 
 impl InternalEvent for BufferEventsReceived {
     fn emit(self) {
-        counter!("buffer_received_events_total", "stage" => self.idx.to_string())
-            .increment(self.count);
-        counter!("buffer_received_bytes_total", "stage" => self.idx.to_string())
-            .increment(self.byte_size);
+        counter!("buffer_received_events_total",
+            "buffer_id" => self.buffer_id.clone(),
+            "stage" => self.idx.to_string()
+        )
+        .increment(self.count);
+
+        counter!("buffer_received_bytes_total",
+            "buffer_id" => self.buffer_id.clone(),
+            "stage" => self.idx.to_string()
+        )
+        .increment(self.byte_size);
 
         let count_delta = i64::try_from(self.count).unwrap_or(i64::MAX);
         let bytes_delta = i64::try_from(self.byte_size).unwrap_or(i64::MAX);
-        update_buffer_gauge(self.idx, count_delta, bytes_delta);
+        update_buffer_gauge(&self.buffer_id, self.idx, count_delta, bytes_delta);
```

Comment:
Since this method owns `self`, and `update_buffer_gauge` wants to have an owned copy of the `buffer_id` for the key, you should be able to pass it as a `String` instead of a reference at the end here and avoid the clone in the callee (same below).